src/main/java/org/csanchez/jenkins/plugins/kubernetes/pod/retention/Reaper.java

```diff
@@ -92,12 +92,18 @@ public static Reaper getInstance() {
 
     @Override
     public void onOnline(Computer c, TaskListener listener) throws IOException, InterruptedException {
-        if (c instanceof KubernetesComputer && activated.compareAndSet(false, true)) {
+        if (c instanceof KubernetesComputer) {
+            maybeActivate();
+        }
+    }
+
+    public void maybeActivate() {
+        if (activated.compareAndSet(false, true)) {
             activate();
         }
     }
 
-    private void activate() {
+    private synchronized void activate() {
```

Comment:
Why?

Reply:
Probably incorrect, but there could be a race condition with `activated` that could lead to multiple activations.

Reply:
How so? https://github.com/jenkinsci/kubernetes-plugin/blob/67600eb6ef178b0e5b6ef35b50fd851ccab5c49b/src/main/java/org/csanchez/jenkins/plugins/kubernetes/pod/retention/Reaper.java#L101 should only ever return true once.

Reply:
Ah, you're right.